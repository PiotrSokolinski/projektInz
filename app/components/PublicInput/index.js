/**
 *
 * PublicInput
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import * as Styled from './styled'

const PublicInput = ({ icon: Icon, label, error, inputProps }) => {
  const isIcon = !!Icon
  return (
    <Styled.InputBox>
      {isIcon && <Icon size="22" />}
      <Styled.Wrapper isIcon={isIcon}>
        <Styled.Input {...inputProps} isIcon={isIcon} />
        <Styled.Label isValue={inputProps.value} isIcon={isIcon}>
          {label}
        </Styled.Label>
      </Styled.Wrapper>
      {error && <Styled.Error color="danger">{error}</Styled.Error>}
    </Styled.InputBox>
  )
}

PublicInput.propTypes = {
  icon: PropTypes.object,
  label: PropTypes.string,
  error: PropTypes.string,
  inputProps: PropTypes.shape({
    type: PropTypes.string,
    name: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
  }),
}

PublicInput.defaultProps = {
  label: '',
  icon: null,
  error: null,
  inputProps: {
    value: '',
    onBlur: () => {},
    onChange: () => {},
  },
}

export default PublicInput
