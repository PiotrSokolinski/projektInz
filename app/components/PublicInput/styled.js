import styled from 'styled-components'
import { Help } from 'react-bulma-components/lib/components/form'
import { Colors } from 'themes'
export const Error = styled(Help)`
  top: -25px;
  position: relative;
`

export const InputBox = styled.div`
  position: relative;
`
export const Input = styled.input`
  width: 100%;
  padding: 10px 0;
  font-size: 16px;
  color: ${Colors.white};
  letter-spacing: 1px;
  margin-bottom: 30px;
  border: none;
  outline: none;
  background: transparent;
  border-bottom: 1px solid ${Colors.white};
  text-indent: ${props => (props.isIcon ? '30px' : '0px')};
`

export const Label = styled.label`
  position: absolute;
  top: ${props => (props.isIcon ? '22px' : '0px')};
  left: ${props => (props.isIcon ? '30px' : '0px')};
  padding: 10px 0;
  font-size: 16px;
  color: ${Colors.white};
  pointer-events: none;
  transition: ${props => (props.isIcon ? '0.5s' : '0.3s')};
  ${props => (props.isValue ? `left: 0; color: ${Colors.cerulean}` : '')};
  ${props => (props.isValue && props.isIcon ? 'top: -15px;' : '')};
  ${props => (props.isValue && !props.isIcon ? 'top: -30px;' : '')};
`

export const Wrapper = styled.div`
  & ${Input}:hover + ${Label} {
    top: ${props => (props.isIcon ? '-15px' : '-30px')};
    left: 0;
    color: ${Colors.cerulean};
  }
`
