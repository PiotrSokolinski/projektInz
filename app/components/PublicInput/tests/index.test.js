/**
 *
 * Tests for PublicInput
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import PublicInput from '../index'

describe('<PublicInput />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<PublicInput />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<PublicInput />)
    expect(firstChild).toMatchSnapshot()
  })
})
