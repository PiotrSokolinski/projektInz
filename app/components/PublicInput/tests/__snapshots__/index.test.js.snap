// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicInput /> Should render and match the snapshot 1`] = `
<div
  class="styled__InputBox-sc-1nwnwj1-1 iARNcM"
>
  <div
    class="styled__Wrapper-sc-1nwnwj1-4 gJixhu"
  >
    <input
      class="styled__Input-sc-1nwnwj1-2 dfHhpa"
      value=""
    />
    <label
      class="styled__Label-sc-1nwnwj1-3 cYwlur"
    />
  </div>
</div>
`;
